mirascope/core/anthropic/stream.py

```diff
@@ -34,9 +40,67 @@ def cost(self) -> float | None:
     def _construct_message_param(
         self, tool_calls: list[ToolUseBlock] | None = None, content: str | None = None
     ) -> MessageParam:
-        """Returns the tool message parameters for tool call results."""
+        """Constructs the message parameter for the assistant."""
+        message_param_tool_calls = []
+        if tool_calls:
+            message_param_tool_calls: list[ToolUseBlockParam] = [
+                tool_call.model_dump()
+                for tool_call in tool_calls  # type: ignore
+            ]
         return {
             "role": "assistant",
-            "content": ([TextBlock(text=content, type="text")] if content else [])
-            + (tool_calls or []),
+            "content": ([TextBlockParam(text=content, type="text")] if content else [])
+            + message_param_tool_calls,
         }
+
+    def construct_call_response(self) -> AnthropicCallResponse:
+        if self.message_param is None:
+            raise ValueError(  # pragma: no cover
+                "No stream response, check if the stream has been consumed."
+            )
+        usage = Usage(input_tokens=0, output_tokens=0)
+        if self.input_tokens:
+            usage.input_tokens = int(self.input_tokens)
+        if self.output_tokens:
+            usage.output_tokens = int(self.output_tokens)
+
+        message_param: list[ContentBlock] = []
```

Comment:
I'm confused what the difference here is between this local `message_param` and the `self.message_param` below. The naming here makes it seem like the stuff below is unnecessary.

I'm assuming we have to do some conversion for some reason, but it's unclear why. Should this just be named `content`?

Reply:
we are converting MessageParam to ContentBlock, from dict to pydantic model

Reply:
right, `list[ContentBlock]` should not be named `message_param` it should be named `content`

Reply:
done